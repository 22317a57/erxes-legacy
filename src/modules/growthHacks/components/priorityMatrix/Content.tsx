import { ContentContainer } from 'modules/growthHacks/styles';
import React from 'react';
import Chart from './Chart';
import Left from './Left';

type Props = {
  growthHacks: any[];
  growthHacksPriorityMatrix: any[];
};

class Content extends React.Component<Props> {
  render() {
    const { growthHacks } = this.props;

    const growthHacksPriorityMatrix = this.props.growthHacksPriorityMatrix.map(
      gh => ({
        y: gh.impact,
        x: gh.ease,
        name: gh.name
      })
    );

    return (
      <ContentContainer>
        <Left growthHacks={growthHacks} />
        <Chart datas={growthHacksPriorityMatrix} />
      </ContentContainer>
    );
  }
}

export default Content;
