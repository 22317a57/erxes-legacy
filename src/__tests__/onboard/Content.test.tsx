import { mount, shallow } from 'enzyme';
import React from 'react';

import Content from '../../modules/onboard/components/Content';

describe('Content component', () => {
  const defaultProps = {
    activeStep: 0
  };

  test('renders Content successfully', () => {
    shallow(<Content {...defaultProps} />);
  });

  test('renders successfully with default value', () => {
    const wrapper = mount(<Content {...defaultProps} />);
    const props = wrapper.props();

    expect(props).toMatchObject(defaultProps);
  });
});
