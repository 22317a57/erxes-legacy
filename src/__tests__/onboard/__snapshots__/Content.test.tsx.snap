// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component snapshot matches 1`] = `null`;
